automatic-config-updates/README.md

```diff
@@ -0,0 +1,67 @@
+# Automatic Configuration Updates
+
+Every now end then, a need to do a backward-incompatible change in `packit.yaml` arises (e.g. renaming a field in the config to a more precise name).
+Currently, we have no convenient way of telling if all users have migrated their configs to the new format (so that the previous one can be removed) which slows down the process of deprecation. Moreover, in a lot of cases, we may be able to do the transformation for the user on our end.
+
+## Possible high-level approaches
+
+There seem to be 2 possible approaches how such updates could work on the top-level (for example the pre-commit bot does both of these):
+
+1. When a user opens a pull request and we detect that the config in the PR can be transformed, we can push a new commit to the pull request.
+2. We could periodically (or make this an action that can be triggered manually, e.g. a script that we run on the side) go over all the currently used configs and search for possible transformations and proactively make a PR to the respective repositories.
+
+A disadvantage of 1) is that it requires user action (in the form of a PR).
+If a project that has Packit enabled doesn't have any PR for a few months, during which we change the config and remove support for the old format, things like triggering `propose-downstream` won't work because the config will be invalid (because there was no PR where we could update it) and unparsable.
+With this in mind, I consider 2) to be a must-have, while 1) to be a nice-to-have.
+
+On the other hand, 1) has the advantage that it can fix the config of the user before it is merged to main if they try to use a deprecated feature, which 2) can't really handle (the config would be merged and then we would create a PR with the correction, so a correct config will end up on the main branch in the end).
+
+## Implementation approach
+
+Regardless of which of the following approaches ends up being used, the process of updating the config will always be the following:
+
+1. Get the config, determine if transformations need to be done
+2. Pull the repository
+3. Make transformations to the config
+4. Create a pull request (or add a commit to a PR) with the transformations
+
+Let's take a closer look at the more "problematic" parts of this pipeline.
+
+### Getting the config
+
+There seem to be three main approaches to getting the config:
+
+1. In case we do the modifications by adding commits to PRs, the config should come from the PR.
+2. Get the list of projects by querying `/api/projects` and then find configs in them.
+3. Add a new database table mapping project to its config content on main.
```

Comment:
YES! We should start collecting these so we can easily access configs of our users.